### PR TITLE
Job: Use generic Set in controller

### PR DESCRIPTION
**Pull request by tenzen-y**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
I replaced deprecated `sets.String` and `sets.Int` with `sets.Set[string]` and `sets.Set[int]` in the Job controller.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/controller/job/job_controller.go [hunk 1]
```diff
@@ -1320,7 +1320,7 @@ func newCondition(conditionType batch.JobConditionType, status v1.ConditionStatu
 }
 
 // getFailJobMessage returns a job failure message if the job should fail with the current counters
-func getFailJobMessage(job *batch.Job, pods []*v1.Pod, uncounted sets.String) *string {
```
Comment (tenzen-y) [inline pkg/controller/job/job_controller.go:1323]: This function doesn't use `uncounted`.

Comment (mimowo) [reply pkg/controller/job/job_controller.go:1323]: Yes, it probably did at some point.

**Pull request merged**